### PR TITLE
SONARKT-261 Improve Rule API script to auto-detect branch to get metadata from.

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

File: buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/GitHubApi.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            }
+            searchResults?.items ?: throw IllegalStateException("Got null response while searching PRs for rule key $ruleKey.")
+        }.map { searchResult ->
+            preparePrDetailsQuery(searchResult.number).getResponseObject<Pr>().let { (response, result) ->
+                val (pr, error) = result
+                error?.let {
+                    System.err.println("Request: ${response.url}")
+                    System.err.println("Response: ${response.data.decodeToString()}")
+                    throw it
+                }
```
Comment (margarita-nedzelska-sonarsource) [inline buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/GitHubApi.kt:36]: maybe this logic is worth being centralized somewhere as it's duplicating lines 23-26

**Review by margarita-nedzelska-sonarsource — COMMENTED** [review]
Overall looks good! Thanks for these improvements.
It just feels a bit complex, maybe adding some comments about what functions do will help. Anyway, I'm okay to merge it as it's our internal tool.

File: buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/GitHubApi.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                if (pr == null) throw IllegalStateException("Got null response while fetching PR #${searchResult.number} details.")
+                pr
```
[the remark this answers is not in the document]
Comment (johann-beleites-sonarsource) [reply buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/GitHubApi.kt:38]: I wrote too much Java :skull:

**Comment by johann-beleites-sonarsource** [comment]
> Overall looks good! Thanks for these improvements. It just feels a bit complex, maybe adding some comments about what functions do will help. Anyway, I'm okay to merge it as it's our internal tool.

Thanks for your comments, some parts were indeed not ideal. In fact, when going through your suggestions, I realized that some error handling can be simplified even further. I think it should be a lot easier to understand now.

**Review by margarita-nedzelska-sonarsource — APPROVED** [review]

**Pull request merged**